src/Language/PureScript/Ide/State.hs

```diff
@@ -181,7 +182,8 @@ populateStage2 = do
 populateStage2STM :: TVar IdeState -> STM ()
 populateStage2STM ref = do
   modules <- s1Modules <$> getStage1STM ref
-  let spans = map (\((P.Module ss _ _ decls _), _) -> M.fromList (concatMap (extractSpans ss) decls)) modules
+  let !spans = map (\((P.Module ss _ _ decls _), _) ->
```

Comment:
What is the type of `spans` here? Forcing it to WHNF probably isn't doing much extra work if it's a list.